apps/api/src/app/groups/groups.service.ts

```diff
@@ -316,21 +363,15 @@ export class GroupsService {
      * Returns a list of groups.
      * @returns List of existing groups.
      */
-    async getAllGroups(): Promise<Group[]> {
-        return this.groupRepository.find({
-            relations: { members: true }
-        })
-    }
+    async getAllGroups(filters?: { adminId: string }): Promise<Group[]> {
+        const where = []
+
+        if (filters?.adminId) {
+            where.push({ adminId: filters.adminId })
+        }
 
-    /**
-     * Returns a list of groups of a specific admin.
-     * @param admin Admin id.
-     * @returns List of admin's existing groups.
-     */
-    async getGroupsByAdmin(admin: string): Promise<Group[]> {
         return this.groupRepository.find({
-            relations: { members: true },
-            where: { adminId: admin }
+            relations: { members: true }
```

Comment:
@saleel It needs the `where` parameter I think

Reply:
True, good catch. Surprisingly ES lint don't catch this for unused variable.
It worked well when I tested because I only had one admin. I have now updated the test case to test with groups created by multiple admins as well.

On a separate note, many of our tests are currently dependent on each other - the changes made by one test affects others. This is not good and can lead to tests show as false positives in some cases.

Reply:
True, we need to improve tests too